### DISCUSSION TITLE
动态路由,应该使用那个请求策略?

### DISCUSSION THREAD

**Discussion by gopower** [alovajs/alova#507] [Q&A]
vue 中动态路由,应该使用那个请求策略?

这样写对吗?

const { data: secondCategoryList, send } = useRequest((id) => getSecondCategoryAPI(id), {
  initialData: {}
})
onMounted(() => {
  send(route.params.id as string)
})

onBeforeRouteUpdate((to) => {
  send(to.params.id as string)
})

**Comment by JOU-amjs** [comment]
`useRequest`在组件中使用是没错的，请说明清楚你要实现的需求我才能更好给出建议哦
